build.gradle

```diff
@@ -46,6 +47,29 @@ ext.compatibilityVersion = '1.7'
 sourceCompatibility = compatibilityVersion
 targetCompatibility = compatibilityVersion
 
+sourceSets {
+    main.java.srcDirs = ['src/main/java'
+                        ,'src/generated/java' ]
+}
+
+compileJava.doFirst {
+ File generatedDir = file("${rootDir}/src/generated/java")
+ if (generatedDir.exists()) {
+  delete generatedDir
+ }
+ generatedDir.mkdirs()
+}
+
+clean.doFirst {
+ File generatedDir = file("${rootDir}/src/generated/java")
+ if (generatedDir.exists()) {
+  delete generatedDir
+ }
+ generatedDir.mkdirs()
+}
```

Comment:
I use netbeans and sometimes intellij. I often don't have a need to see these classes but when I do it's easy enough through both IDE's to look within the filesystem view.

Furthermore the AutoValue classes are already available within the gradle `build/classes/*` folders. Is that not enough?